consoles/sshVirtsh.pm

```diff
@@ -496,6 +499,33 @@ sub resume ($self) {
 
 sub get_remote_vmm ($self) { $bmwqemu::vars{VMWARE_REMOTE_VMM} // '' }
 
+sub _encode_config ($self, $file) {
+    unless (!!$file) {
+        croak "Please provide a configuration file to encode!";
```

Comment:
Maybe it would make sense if this function would get only the key and read `$bmwqemu::vars{$setting_key}`. Then it could tell the user what concrete variable is missing. Otherwise the error message could also just list all relevant variables.

Reply:
That crossed my mind too. The problem over here is that in case of cloud-init we need to encode user-data and meta-data. So the function would need to return 2 strings. Also the variable as of now is not single but depends on the configuration tool that we want to use (combustion, ignition, cloud-init). I will keep this open and rethink what would be a better approach

Reply:
You can also just pass the name of the variable as extra argument.